### PR TITLE
do ugly things to wire a cloudconfig file

### PR DESCRIPTION
**Pull request by deads2k**
Might fix https://bugzilla.redhat.com/show_bug.cgi?id=1454601 

This punches through a couple layers to build an options object twice in order to plumb a value through an admission wiring step before admission can logically be completed to hopefully make an admission plugin work.

@eparis

### CODE REVIEW

**Comment by deads2k** [comment]
[test]

No tests for this, so it probably works.

File: pkg/cmd/server/origin/master_config.go [hunk 1]
```diff
@@ -291,10 +292,24 @@ func BuildMasterConfig(options configapi.MasterConfig) (*MasterConfig, error) {
...
+		cloudConfig, err = ioutil.ReadFile(kubeOptions.CloudProvider.CloudConfigFile)
+		if err != nil {
+			return nil, err
+			glog.Fatalf("Error reading from cloud configuration file %s: %#v", kubeOptions.CloudProvider.CloudConfigFile, err)
+		}
+	}
 	// note: we are passing a combined quota registry here...
-	kubePluginInitializer := kadmission.NewPluginInitializer(privilegedLoopbackKubeClientsetInternal, internalkubeInformerFactory, nil, nil, quotaRegistry)
+	kubePluginInitializer := kadmission.NewPluginInitializer(privilegedLoopbackKubeClientsetInternal, internalkubeInformerFactory, authorizer, cloudConfig, quotaRegistry)
 	originAdmission, kubeAdmission, err := buildAdmissionChains(options, privilegedLoopbackKubeClientsetInternal, pluginInitializer, kubePluginInitializer)
```
Comment (gnufied) [inline pkg/cmd/server/origin/master_config.go]: Why are we passing `authorizer` here now?

**Comment by gnufied** [comment]
I have tested this to work on a multizone GCE cluster. Also added a card for writing a e2e test - https://trello.com/c/Ul2Ldjjn/508-write-an-e2e-test-for-multizone-fix

**Comment by liggitt** [comment]
is this what the upstream admission PR will improve? are you planning to pick that, or carry this until we get to 1.7?

**Comment by deads2k** [comment]
> is this what the upstream admission PR will improve? are you planning to pick that, or carry this until we get to 1.7?

Well, I thought we had a controller problem, not an admission one.  I think we need to refactor the upstream admission plugin to use runtime information instead of depending on files on disk that deep into the init structure, but that would be 1.8 I think.

**Comment by deads2k** [comment]
There was a request to merge before the sprint opened, so I'll take that as lgtm.  Doing it now [merge]

**Comment by gnufied** [comment]
Turns out we already have an e2e test for statically provisioned PVs - https://github.com/kubernetes/kubernetes/pull/42829 . Although, it was added kind of recently.

**Comment by deads2k** [comment]
https://ci.openshift.redhat.com/jenkins/job/merge_pull_request_origin/913/ flake https://github.com/openshift/origin/issues/14396


re[merge]

**Pull request merged**